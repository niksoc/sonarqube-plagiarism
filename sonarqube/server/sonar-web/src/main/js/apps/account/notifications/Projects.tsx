/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { groupBy, sortBy, uniqBy } from 'lodash';
import * as React from 'react';
import { Button } from 'sonar-ui-common/components/controls/buttons';
import SearchBox from 'sonar-ui-common/components/controls/SearchBox';
import { translate } from 'sonar-ui-common/helpers/l10n';
import ProjectModal from './ProjectModal';
import ProjectNotifications from './ProjectNotifications';

export interface Props {
  addNotification: (n: T.Notification) => void;
  channels: string[];
  notifications: T.Notification[];
  removeNotification: (n: T.Notification) => void;
  types: string[];
}

const THRESHOLD_COLLAPSED = 3;

interface State {
  addedProjects: T.NotificationProject[];
  search: string;
  showModal: boolean;
}

function isNotificationProject(project: {
  project?: string;
  projectName?: string;
}): project is T.NotificationProject {
  return project.project !== undefined && project.projectName !== undefined;
}

export default class Projects extends React.PureComponent<Props, State> {
  state: State = {
    addedProjects: [],
    search: '',
    showModal: false
  };

  filterSearch = (project: T.NotificationProject, search: string) => {
    return project.projectName && project.projectName.toLowerCase().includes(search);
  };

  handleAddProject = (project: T.NotificationProject) => {
    this.setState(state => {
      return {
        addedProjects: [...state.addedProjects, project]
      };
    });
  };

  handleSearch = (search = '') => {
    this.setState({ search: search.toLowerCase() });
  };

  handleSubmit = (selectedProject: T.NotificationProject) => {
    if (selectedProject) {
      this.handleAddProject(selectedProject);
    }

    this.closeModal();
  };

  closeModal = () => {
    this.setState({ showModal: false });
  };

  openModal = () => {
    this.setState({ showModal: true });
  };

  removeNotification = (removed: T.Notification, allProjects: T.NotificationProject[]) => {
    const projectToRemove = allProjects.find(p => p.project === removed.project);
    if (projectToRemove) {
      this.handleAddProject(projectToRemove);
    }

    this.props.removeNotification(removed);
  };

  render() {
    const { notifications } = this.props;
    const { addedProjects, search } = this.state;

    const projects = uniqBy(notifications, project => project.project).filter(
      isNotificationProject
    ) as T.NotificationProject[];
    const notificationsByProject = groupBy(notifications, n => n.project);
    const allProjects = uniqBy([...addedProjects, ...projects], project => project.project);
    const filteredProjects = sortBy(allProjects, 'projectName').filter(p =>
      this.filterSearch(p, search)
    );
    const shouldBeCollapsed = Object.keys(notificationsByProject).length > THRESHOLD_COLLAPSED;

    return (
      <section className="boxed-group" data-test="account__project-notifications">
        <div className="boxed-group-inner">
          <div className="page-actions">
            <Button onClick={this.openModal}>
              <span data-test="account__add-project-notification">
                {translate('my_profile.per_project_notifications.add')}
              </span>
            </Button>
          </div>

          <h2>{translate('my_profile.per_project_notifications.title')}</h2>
        </div>

        {this.state.showModal && (
          <ProjectModal
            addedProjects={allProjects}
            closeModal={this.closeModal}
            onSubmit={this.handleSubmit}
          />
        )}

        <div className="boxed-group-inner">
          {allProjects.length === 0 && (
            <div className="note">{translate('my_account.no_project_notifications')}</div>
          )}

          {allProjects.length > 0 && (
            <div className="big-spacer-bottom">
              <SearchBox
                onChange={this.handleSearch}
                placeholder={translate('search.search_for_projects')}
              />
            </div>
          )}

          {filteredProjects.map(project => {
            const collapsed = addedProjects.find(p => p.project === project.project)
              ? false
              : shouldBeCollapsed;
            return (
              <ProjectNotifications
                addNotification={this.props.addNotification}
                channels={this.props.channels}
                collapsed={collapsed}
                key={project.project}
                notifications={notificationsByProject[project.project] || []}
                project={project}
                removeNotification={n => this.removeNotification(n, allProjects)}
                types={this.props.types}
              />
            );
          })}
        </div>
      </section>
    );
  }
}
