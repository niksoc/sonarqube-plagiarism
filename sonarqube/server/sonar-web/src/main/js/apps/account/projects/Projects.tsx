/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import ListFooter from 'sonar-ui-common/components/controls/ListFooter';
import { translate } from 'sonar-ui-common/helpers/l10n';
import ProjectCard from './ProjectCard';

interface Props {
  loading: boolean;
  loadMore: () => void;
  projects: T.MyProject[];
  total?: number;
}

export default function Projects(props: Props) {
  const { projects } = props;

  return (
    <div id="account-projects">
      {projects.length === 0 ? (
        <div className="js-no-results">{translate('my_account.projects.no_results')}</div>
      ) : (
        <p>{translate('my_account.projects.description')}</p>
      )}

      {projects.length > 0 && (
        <ul className="account-projects-list">
          {projects.map(project => (
            <li key={project.key}>
              <ProjectCard project={project} />
            </li>
          ))}
        </ul>
      )}

      {projects.length > 0 && (
        <ListFooter
          count={projects.length}
          loadMore={props.loadMore}
          ready={!props.loading}
          total={props.total || 0}
        />
      )}
    </div>
  );
}
