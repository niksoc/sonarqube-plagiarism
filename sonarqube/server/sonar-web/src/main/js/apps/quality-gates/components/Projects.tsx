/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { find, without } from 'lodash';
import * as React from 'react';
import SelectList, {
  SelectListFilter,
  SelectListSearchParams
} from 'sonar-ui-common/components/controls/SelectList';
import { translate } from 'sonar-ui-common/helpers/l10n';
import {
  associateGateWithProject,
  dissociateGateWithProject,
  searchProjects
} from '../../../api/quality-gates';

interface Props {
  canEdit?: boolean;
  organization?: string;
  qualityGate: T.QualityGate;
}

interface State {
  needToReload: boolean;
  lastSearchParams?: SelectListSearchParams;
  projects: Array<{ id: string; key: string; name: string; selected: boolean }>;
  projectsTotalCount?: number;
  selectedProjects: string[];
}

export default class Projects extends React.PureComponent<Props, State> {
  mounted = false;

  constructor(props: Props) {
    super(props);

    this.state = {
      needToReload: false,
      projects: [],
      selectedProjects: []
    };
  }

  componentDidMount() {
    this.mounted = true;
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  fetchProjects = (searchParams: SelectListSearchParams) =>
    searchProjects({
      gateId: this.props.qualityGate.id,
      organization: this.props.organization,
      page: searchParams.page,
      pageSize: searchParams.pageSize,
      query: searchParams.query !== '' ? searchParams.query : undefined,
      selected: searchParams.filter
    }).then(data => {
      if (this.mounted) {
        this.setState(prevState => {
          const more = searchParams.page != null && searchParams.page > 1;

          const projects = more ? [...prevState.projects, ...data.results] : data.results;
          const newSelectedProjects = data.results
            .filter(project => project.selected)
            .map(project => project.id);
          const selectedProjects = more
            ? [...prevState.selectedProjects, ...newSelectedProjects]
            : newSelectedProjects;

          return {
            lastSearchParams: searchParams,
            needToReload: false,
            projects,
            projectsTotalCount: data.paging.total,
            selectedProjects
          };
        });
      }
    });

  handleSelect = (id: string) =>
    associateGateWithProject({
      gateId: this.props.qualityGate.id,
      organization: this.props.organization,
      projectId: id
    }).then(() => {
      if (this.mounted) {
        this.setState(prevState => ({
          needToReload: true,
          selectedProjects: [...prevState.selectedProjects, id]
        }));
      }
    });

  handleUnselect = (id: string) =>
    dissociateGateWithProject({
      gateId: this.props.qualityGate.id,
      organization: this.props.organization,
      projectId: id
    }).then(() => {
      if (this.mounted) {
        this.setState(prevState => ({
          needToReload: true,
          selectedProjects: without(prevState.selectedProjects, id)
        }));
      }
    });

  renderElement = (id: string): React.ReactNode => {
    const project = find(this.state.projects, { id });
    return (
      <div className="select-list-list-item">
        {project === undefined ? (
          id
        ) : (
          <>
            {project.name}
            <br />
            <span className="note">{project.key}</span>
          </>
        )}
      </div>
    );
  };

  render() {
    return (
      <SelectList
        elements={this.state.projects.map(project => project.id)}
        elementsTotalCount={this.state.projectsTotalCount}
        labelAll={translate('quality_gates.projects.all')}
        labelSelected={translate('quality_gates.projects.with')}
        labelUnselected={translate('quality_gates.projects.without')}
        needToReload={
          this.state.needToReload &&
          this.state.lastSearchParams &&
          this.state.lastSearchParams.filter !== SelectListFilter.All
        }
        onSearch={this.fetchProjects}
        onSelect={this.handleSelect}
        onUnselect={this.handleUnselect}
        readOnly={!this.props.canEdit}
        renderElement={this.renderElement}
        selectedElements={this.state.selectedProjects}
        withPaging={true}
      />
    );
  }
}
