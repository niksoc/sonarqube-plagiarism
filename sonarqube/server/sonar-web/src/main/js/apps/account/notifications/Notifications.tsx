/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { partition } from 'lodash';
import * as React from 'react';
import { Helmet } from 'react-helmet-async';
import { Alert } from 'sonar-ui-common/components/ui/Alert';
import DeferredSpinner from 'sonar-ui-common/components/ui/DeferredSpinner';
import { translate } from 'sonar-ui-common/helpers/l10n';
import {
  withNotifications,
  WithNotificationsProps
} from '../../../components/hoc/withNotifications';
import GlobalNotifications from './GlobalNotifications';
import Projects from './Projects';

export function Notifications(props: WithNotificationsProps) {
  const {
    addNotification,
    channels,
    globalTypes,
    loading,
    notifications,
    perProjectTypes,
    removeNotification
  } = props;

  const [globalNotifications, projectNotifications] = partition(notifications, n => !n.project);

  return (
    <div className="account-body account-container">
      <Helmet defer={false} title={translate('my_account.notifications')} />
      <Alert variant="info">{translate('notification.dispatcher.information')}</Alert>
      <DeferredSpinner loading={loading}>
        {notifications && (
          <>
            <GlobalNotifications
              addNotification={addNotification}
              channels={channels}
              notifications={globalNotifications}
              removeNotification={removeNotification}
              types={globalTypes}
            />
            <Projects
              addNotification={addNotification}
              channels={channels}
              notifications={projectNotifications}
              removeNotification={removeNotification}
              types={perProjectTypes}
            />
          </>
        )}
      </DeferredSpinner>
    </div>
  );
}

export default withNotifications(Notifications);
