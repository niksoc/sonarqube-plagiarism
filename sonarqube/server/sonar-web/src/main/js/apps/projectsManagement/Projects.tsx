/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as classNames from 'classnames';
import * as React from 'react';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { Project } from '../../api/components';
import ProjectRow from './ProjectRow';

interface Props {
  currentUser: Pick<T.LoggedInUser, 'login'>;
  onProjectDeselected: (project: string) => void;
  onProjectSelected: (project: string) => void;
  organization: T.Organization;
  projects: Project[];
  ready?: boolean;
  selection: string[];
}

export default class Projects extends React.PureComponent<Props> {
  onProjectCheck = (project: Project, checked: boolean) => {
    if (checked) {
      this.props.onProjectSelected(project.key);
    } else {
      this.props.onProjectDeselected(project.key);
    }
  };

  render() {
    return (
      <div className="boxed-group boxed-group-inner">
        <table
          className={classNames('data', 'zebra', { 'new-loading': !this.props.ready })}
          id="projects-management-page-projects">
          <thead>
            <tr>
              <th />
              <th>{translate('name')}</th>
              <th />
              <th>{translate('key')}</th>
              <th className="thin nowrap text-right">{translate('last_analysis')}</th>
              <th />
            </tr>
          </thead>
          <tbody>
            {this.props.projects.map(project => (
              <ProjectRow
                currentUser={this.props.currentUser}
                key={project.key}
                onProjectCheck={this.onProjectCheck}
                organization={this.props.organization && this.props.organization.key}
                project={project}
                selected={this.props.selection.includes(project.key)}
              />
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
