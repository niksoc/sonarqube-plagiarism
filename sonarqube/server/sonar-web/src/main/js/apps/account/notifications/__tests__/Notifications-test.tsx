/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import GlobalNotifications from '../GlobalNotifications';
import { Notifications } from '../Notifications';
import Projects from '../Projects';

it('should render correctly', () => {
  expect(shallowRender({ loading: true })).toMatchSnapshot();
  expect(shallowRender()).toMatchSnapshot();
  expect(shallowRender({ notifications: [] })).toMatchSnapshot();
});

it('should add and remove global notifications', () => {
  const addNotification = jest.fn();
  const removeNotification = jest.fn();
  const notification = { channel: 'channel2', type: 'type-global' };
  const wrapper = shallowRender({ addNotification, removeNotification });

  wrapper
    .find(GlobalNotifications)
    .props()
    .addNotification(notification);
  expect(addNotification).toBeCalledWith(notification);

  wrapper
    .find(GlobalNotifications)
    .props()
    .removeNotification(notification);
  expect(removeNotification).toBeCalledWith(notification);
});

it('should add and remove project notification', () => {
  const addNotification = jest.fn();
  const removeNotification = jest.fn();
  const notification = {
    channel: 'channel2',
    type: 'type-common',
    project: 'qux'
  };
  const wrapper = shallowRender({ addNotification, removeNotification });

  wrapper
    .find(Projects)
    .props()
    .addNotification(notification);
  expect(addNotification).toBeCalledWith(notification);

  wrapper
    .find(Projects)
    .props()
    .removeNotification(notification);
  expect(removeNotification).toBeCalledWith(notification);
});

function shallowRender(props = {}) {
  return shallow(
    <Notifications
      addNotification={jest.fn()}
      channels={['channel1', 'channel2']}
      globalTypes={['type-global', 'type-common']}
      loading={false}
      notifications={[
        {
          channel: 'channel1',
          type: 'type-global',
          project: 'foo',
          projectName: 'Foo',
          organization: 'org'
        },
        {
          channel: 'channel1',
          type: 'type-common',
          project: 'bar',
          projectName: 'Bar',
          organization: 'org'
        },
        {
          channel: 'channel2',
          type: 'type-common',
          project: 'qux',
          projectName: 'Qux',
          organization: 'org'
        }
      ]}
      perProjectTypes={['type-common']}
      removeNotification={jest.fn()}
      {...props}
    />
  );
}
