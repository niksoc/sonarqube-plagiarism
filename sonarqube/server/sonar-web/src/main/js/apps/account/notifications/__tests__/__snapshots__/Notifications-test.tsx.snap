// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="account-body account-container"
>
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="my_account.notifications"
  />
  <Alert
    variant="info"
  >
    notification.dispatcher.information
  </Alert>
  <DeferredSpinner
    loading={true}
    timeout={100}
  >
    <GlobalNotifications
      addNotification={[MockFunction]}
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      notifications={Array []}
      removeNotification={[MockFunction]}
      types={
        Array [
          "type-global",
          "type-common",
        ]
      }
    />
    <Projects
      addNotification={[MockFunction]}
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      notifications={
        Array [
          Object {
            "channel": "channel1",
            "organization": "org",
            "project": "foo",
            "projectName": "Foo",
            "type": "type-global",
          },
          Object {
            "channel": "channel1",
            "organization": "org",
            "project": "bar",
            "projectName": "Bar",
            "type": "type-common",
          },
          Object {
            "channel": "channel2",
            "organization": "org",
            "project": "qux",
            "projectName": "Qux",
            "type": "type-common",
          },
        ]
      }
      removeNotification={[MockFunction]}
      types={
        Array [
          "type-common",
        ]
      }
    />
  </DeferredSpinner>
</div>
`;

exports[`should render correctly 2`] = `
<div
  className="account-body account-container"
>
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="my_account.notifications"
  />
  <Alert
    variant="info"
  >
    notification.dispatcher.information
  </Alert>
  <DeferredSpinner
    loading={false}
    timeout={100}
  >
    <GlobalNotifications
      addNotification={[MockFunction]}
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      notifications={Array []}
      removeNotification={[MockFunction]}
      types={
        Array [
          "type-global",
          "type-common",
        ]
      }
    />
    <Projects
      addNotification={[MockFunction]}
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      notifications={
        Array [
          Object {
            "channel": "channel1",
            "organization": "org",
            "project": "foo",
            "projectName": "Foo",
            "type": "type-global",
          },
          Object {
            "channel": "channel1",
            "organization": "org",
            "project": "bar",
            "projectName": "Bar",
            "type": "type-common",
          },
          Object {
            "channel": "channel2",
            "organization": "org",
            "project": "qux",
            "projectName": "Qux",
            "type": "type-common",
          },
        ]
      }
      removeNotification={[MockFunction]}
      types={
        Array [
          "type-common",
        ]
      }
    />
  </DeferredSpinner>
</div>
`;

exports[`should render correctly 3`] = `
<div
  className="account-body account-container"
>
  <Helmet
    defer={false}
    encodeSpecialCharacters={true}
    title="my_account.notifications"
  />
  <Alert
    variant="info"
  >
    notification.dispatcher.information
  </Alert>
  <DeferredSpinner
    loading={false}
    timeout={100}
  >
    <GlobalNotifications
      addNotification={[MockFunction]}
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      notifications={Array []}
      removeNotification={[MockFunction]}
      types={
        Array [
          "type-global",
          "type-common",
        ]
      }
    />
    <Projects
      addNotification={[MockFunction]}
      channels={
        Array [
          "channel1",
          "channel2",
        ]
      }
      notifications={Array []}
      removeNotification={[MockFunction]}
      types={
        Array [
          "type-common",
        ]
      }
    />
  </DeferredSpinner>
</div>
`;
